modules/slms_3/1.0/slms_3.smk

```diff
@@ -0,0 +1,451 @@
+#!/usr/bin/env snakemake
+
+
+##### ATTRIBUTION #####
+
+
+# Original Author:  Laura Hilton
+# Module Author:    Laura Hilton
+# Contributors:     N/A
+
+
+##### SETUP #####
+
+
+# Import package oncopipe and inspect packages. 
+import oncopipe as op
+import inspect
+
+# Check that the oncopipe dependency is up-to-date. Add all the following lines to any module that uses new features in oncopipe
+min_oncopipe_version="1.0.11"
+import pkg_resources
+try:
+    from packaging import version
+except ModuleNotFoundError:
+    sys.exit("The packaging module dependency is missing. Please install it ('pip install packaging') and ensure you are using the most up-to-date oncopipe version")
+
+# To avoid this we need to add the "packaging" module as a dependency for LCR-modules or oncopipe
+
+current_version = pkg_resources.get_distribution("oncopipe").version
+if version.parse(current_version) < version.parse(min_oncopipe_version):
+    print('\x1b[0;31;40m' + f'ERROR: oncopipe version installed: {current_version}' + '\x1b[0m')
+    print('\x1b[0;31;40m' + f"ERROR: This module requires oncopipe version >= {min_oncopipe_version}. Please update oncopipe in your environment" + '\x1b[0m')
+    sys.exit("Instructions for updating to the current version of oncopipe are available at https://lcr-modules.readthedocs.io/en/latest/ (use option 2)")
+
+# End of dependency checking section 
+
+
+# Setup module and store module-specific configuration in `CFG_SLMS3`
+# `CFG_SLMS3` is a shortcut to `config["lcr-modules"]["slms_3"]`
+# This is not assigned to `CFG` as in other modules because the sub-modules here will use `CFG`. 
+CFG_SLMS3 = op.setup_module(
+    name = "slms_3",
+    version = "1.0",
+    subdirectories = ["inputs", "strelka_gnomad", "lofreq_gnomad", "strelka_lofreq_union", "sage_gnomad", "mutect2_depth_filt", "union", "outputs"],
+)
+
+# Ensure each of the submodule versions meets requirements. 
+
+assert float(CFG_SLMS3["module_versions"]["manta"]) >= 2.0, (
+    f"The current manta module version is {CFG_SLMS3['module_versions']['manta']}. "
+    "SLMS-3 requires manta module version 2.0 or higher. "
+)
+
+assert float(CFG_SLMS3["module_versions"]["starfish"]) >= 2.0, (
+    f"The current starfish module version is {CFG_SLMS3['module_versions']['starfish']}. "
+    "SLMS-3 requires starfish module version 2.0 or higher. "
+)
+
+assert float(CFG_SLMS3["module_versions"]["mutect2"]) >= 2.0, (
+    f"The current mutect2 module version is {CFG_SLMS3['module_versions']['mutect2']}. "
+    "SLMS-3 requires mutect2 module version 2.0 or higher. "
+)
+
+assert float(CFG_SLMS3["module_versions"]["strelka"]) >= 1.1, (
+    f"The current strelka module version is {CFG_SLMS3['module_versions']['strelka']}. "
+    "SLMS-3 requires strelka module version 1.1 or higher. "
+)
+
```

Comment:
I thought I commented on this topic but now I can't find the original comment. What are your thoughts on defaulting to the latest version of all these modules so the user (and developers) won't need to keep updating the config any time one of these modules gets a version bump? If it could be done automatically I imagine it would reduce the complexity a lot and simplify the config.

Reply:
My hesitation is that someone might have run an older version of one of the tools and wouldn't want to have to run it again, as I did for everything in icgc_dart which had been run through manta-2.2 instead of 2.3. This is a situation where I thin unintentional updates to the config could have far-reaching consequences for how many things get re-run.

Reply:
That sounds like a very good reason to keep it as-is.